pkgs/development/libraries/mesa/default.nix

```diff
@@ -37,9 +41,15 @@ with lib;
 let
   # Release calendar: https://www.mesa3d.org/release-calendar.html
   # Release frequency: https://www.mesa3d.org/releasing.html#schedule
-  version = "22.2.3";
+  version = "22.3.0";
   branch  = versions.major version;
 
+  rust-bindgen' = rust-bindgen.override {
+    rust-bindgen-unwrapped = rust-bindgen.unwrapped.override {
+      clang = llvmPackages.clang;
+    };
+  };
```

Comment:
Can you please explain why this is necessary? Sorry, I don't understand it.

Reply:
The default bindgen is wrapped with libclang 11 which is too old for building rusticl.

Reply:
Maybe the default `rust-bindgen` is worth bumping to newer clang?  (in a different PR, preferably)  Maintainers: @JohnTitor, @Ralith.

Reply:
Unfortunately the proper solution here is probably to version bindgen (if not rust), as different packages will want a different LLVM in the closure.

Reply:
to make this a little less ugly I would suggest to add an easier way to overwrite clang for rust-bindgen.